packages/odyssey-react-mui/src/labs/DataComponents/StackContent.tsx

```diff
@@ -0,0 +1,247 @@
+import { RowActions } from "./RowActions";
+import { StackCard } from "./StackCard";
+import { StackLayout, StackProps, UniversalProps } from "./componentTypes";
+
+export type StackContentProps = {
+  currentLayout: StackLayout;
+  data: MRT_RowData[];
+  draggingRow?: MRT_Row<MRT_RowData> | null;
+  emptyState: ReactNode;
+  getRowId: UniversalProps["getRowId"];
+  hasRowReordering: UniversalProps["hasRowReordering"];
+  hasRowSelection: UniversalProps["hasRowSelection"];
+  isEmpty?: boolean;
+  isLoading: boolean;
+  isNoResults?: boolean;
+  isRowReorderingDisabled?: boolean;
+  onReorderRows: UniversalProps["onReorderRows"];
+  pagination: { pageIndex: number; pageSize: number };
+  rowReorderingUtilities: {
+    dragHandleStyles: CSSObject;
+    dragHandleText: {
+      title: string;
+      "aria-label": string;
+    };
+    draggableTableBodyRowClassName: ({
+      currentRowId,
+      draggingRowId,
+      hoveredRowId,
+    }: {
+      currentRowId: string;
+      draggingRowId?: string;
+      hoveredRowId?: string;
+    }) => string | undefined;
+    handleDragHandleKeyDown: ({
+      table,
+      row,
+      event,
+    }: {
+      table: MRT_TableInstance<MRT_RowData>;
+      row: MRT_Row<MRT_RowData>;
+      event: React.KeyboardEvent<HTMLButtonElement>;
+    }) => void;
+    handleDragHandleOnDragCapture: (
+      table: MRT_TableInstance<MRT_RowData>,
+    ) => void;
+    handleDragHandleOnDragEnd: (table: MRT_TableInstance<MRT_RowData>) => void;
+    resetDraggingAndHoveredRow: (table: MRT_TableInstance<MRT_RowData>) => void;
+    updateRowOrder: ({
+      rowId,
+      newRowIndex,
+    }: {
+      rowId: string;
+      newRowIndex: number;
+    }) => void;
+  };
+  rowSelection: MRT_RowSelectionState;
+  setRowSelection: Dispatch<SetStateAction<MRT_RowSelectionState>>;
+  stackOptions: StackProps;
+  totalRows: UniversalProps["totalRows"];
+};
+
+const StackContent = ({
+  currentLayout,
+  data,
+  emptyState,
+  hasRowReordering,
+  hasRowSelection,
+  isEmpty,
+  isLoading,
+  isNoResults,
+  isRowReorderingDisabled,
+  onReorderRows,
+  pagination,
+  rowReorderingUtilities,
+  rowSelection,
+  setRowSelection,
+  stackOptions,
+  totalRows,
+}: StackContentProps) => {
+  const handleRowSelectionChange = useCallback(
+    (row: MRT_RowData) => {
+      setRowSelection((prev) => {
+        const newSelection = { ...prev };
+        if (newSelection[row.id]) {
+          delete newSelection[row.id];
+        } else {
+          newSelection[row.id] = true;
+        }
+        return newSelection;
+      });
+    },
+    [setRowSelection],
+  );
+
+  const renderDetailPanelProp = stackOptions.renderDetailPanel;
+
+  const renderDetailPanel = useCallback(
+    (row: MRT_RowData) => {
+      return renderDetailPanelProp?.({ row });
+    },
+    [renderDetailPanelProp],
+  );
+
+  const { updateRowOrder } = rowReorderingUtilities;
+
+  const odysseyDesignTokens = useOdysseyDesignTokens();
```

Comment:
Can we add the design tokens context consumer to the top of the component? Just to keep that as a consistent pattern. It doesn't need to be this far down necessarily, but I do like that it's around the components we're creating.

Still, I don't think we should be creating styled components inline like this. I believe theres a cost to doing so, and React can't properly memoize them.

I'm not 100% sure though, at first glance, my opinion is that you'd have to memoize these styled components with `useMemo`.